### PR TITLE
[READY] fixes roles that shouldn't be creepy being creepy and the jealousy objective asking you to assassinate the obsession + missed sanity

### LINKED ISSUE #42379: anyone can become a creep it seems
https://i.gyazo.com/d6a73eb460f80c0710c8d0d5efb66e07.png

### LINKED ISSUE #42390: creeps are getting an assassinate objective for their obsession as well as the murder one
I noticed this from #42379, i've been trying to figure out why though.

Assumingly this is the jealousy objective, the obsession murder objective is always at the end of the objective list.

https://github.com/tgstation/tgstation/blob/0f4ac811e8b7b35c04f45df705a1713982f5eaf8/code/modules/antagonists/creep/creep.dm#L146-L154

It's not choosing a new target, or updating the explanation text. i'm assuming it's runtiming somewhere!

### PR DESCRIPTION
**Pull request by tralezab**
fixes #42379 
fixes #42390 
and also a ton of other improvements and fixes I found while fixing the above, it's fantastic

image

I would like to note that the fact that the syndicate observer on the asteroid became obsessed with one person on the station, and managed to get them killed to stop themselves from going insane. based storytelling videogame

### CODE REVIEW

File: code/modules/events/creep_awakening.dm [hunk 1]
```diff
@@ -13,7 +13,7 @@
 			continue
 		if(H.stat == DEAD)
 			continue
-		if(!H.mind.assigned_role || H.mind.assigned_role in GLOB.exp_specialmap[EXP_TYPE_SPECIAL] || H.mind.assigned_role in GLOB.exp_specialmap[EXP_TYPE_ANTAG]) //prevents ashwalkers falling in love with crewmembers they never met
+		if(!H.mind.assigned_role || !H.mind.assigned_role in GLOB.station_positions || H.mind.assigned_role in GLOB.nonhuman_positions) //only station jobs sans nonhuman roles, prevents ashwalkers falling in love with crewmembers they never met
```
Comment (tralezab) [inline code/modules/events/creep_awakening.dm]: this first check might be redundant, can let me know if it's even needed or if i can cut it

File: code/modules/antagonists/creep/creep.dm [hunk 2]
```diff
@@ -175,8 +175,9 @@
 	for(var/mob/living/carbon/human/H in GLOB.alive_mob_list)
 		if(!H.mind)
 			continue
-		if(!H.mind.assigned_role || H == oldmind.current || H.mind.has_antag_datum(/datum/antagonist/creep)) //the jealousy target has to have a job, and not be the obsession or creep.
+		if(!H.mind.assigned_role in GLOB.station_positions || H == oldmind.current || H.mind.has_antag_datum(/datum/antagonist/creep)) //the jealousy target has to have a job, and not be the obsession or creep.
```
Comment (Cyberboss) [inline code/modules/antagonists/creep/creep.dm]:
```suggestion
		if(!(H.mind.assigned_role in GLOB.station_positions) || H == oldmind.current || H.mind.has_antag_datum(/datum/antagonist/creep)) //the jealousy target has to have a job, and not be the obsession or creep.
```

File: code/modules/events/creep_awakening.dm (hunk 1 above)
Comment (Cyberboss) [inline code/modules/events/creep_awakening.dm]:
```suggestion
		if(!H.mind.assigned_role || (!H.mind.assigned_role in GLOB.station_positions) || H.mind.assigned_role in GLOB.nonhuman_positions) //only station jobs sans nonhuman roles, prevents ashwalkers falling in love with crewmembers they never met
```

File: code/modules/jobs/jobs.dm [hunk 3]
```diff
@@ -1,3 +1,38 @@
+GLOBAL_LIST_INIT(station_positions, list(
```
Comment (Cyberboss) [inline code/modules/jobs/jobs.dm]: This list HAS to already exist somewhere

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/jobs/jobs.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/jobs/jobs.dm]: I'll take a second look

File: code/modules/events/creep_awakening.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/events/creep_awakening.dm]: Er let me do this with the ! outside the parentheses

**Comment by bgobandit** [comment]
> I would like to note that the fact that the syndicate observer on the asteroid became obsessed with one person on the station, and managed to get them killed to stop themselves from going insane. based storytelling videogame

feature, haven't you ever heard of celebrity stalkers

**Comment by tralezab** [comment]
isn't that so fucking cool?

File: code/modules/jobs/jobs.dm (hunk 3 above)
Comment (tralezab) [reply code/modules/jobs/jobs.dm]: i swear, this doesn't exist somehow

Comment (Jordie0608) [reply code/modules/jobs/jobs.dm]: This should work https://github.com/tgstation/tgstation/blob/a89c5aee03f3660d31715f268b192270e90abeae/code/controllers/subsystem/job.dm#L7

Comment (tralezab) [reply code/modules/jobs/jobs.dm]: i'll test it tonight to make sure, thanks for the help

Comment (tralezab) [reply code/modules/jobs/jobs.dm]: wow, this is great. I should really refactor a lot of the role checking code to work with the job subsystem

**Pull request merged**